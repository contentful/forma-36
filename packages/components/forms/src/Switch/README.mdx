---
title: 'Switch'
type: 'component'
status: 'stable'
section: 'formComponents'
slug: /components/switch/
github: 'https://github.com/contentful/forma-36/tree/master/packages/components/forms/src/Switch'
storybook: 'https://v4-f36-storybook.netlify.app/?path=/story/form-elements-switch--basic'
typescript: ./Switch.tsx
---

Switch is a control that is used to quickly switch between two possible states. Switch works like a physical switch that allows users to turn things on or off, like a light switch.

## How to use Switch

- use when a setting requires an on/off or show/hide function to display the results
- use when user needs to perform instantaneous actions that do not need a review or confirmation
- use when user is toggling independent features or behaviors

### Using as a controlled input

For using the Switch as a controlled input, you need to:

- Pass the `isChecked` property, with this property it will already be a controlled component;
- Pass a `onChange` handler, so you can use it to update the value of `isChecked`;

Setting the `isChecked` will already define it as a controlled input.

```jsx static=true
const ExampleControlled = () => {
  const [switchState, setSwitchState] = useState(false);

  return (
    <Switch
      name="allow-cookies-controlled"
      id="allow-cookies-controlled"
      isChecked={switchState}
      onChange={() => setSwitchState((prevState) => !prevState)}
    >
      Allow cookies
    </Switch>
  );
};
```

### Using as an uncontrolled input

You can use the Switch as an uncontrolled input, for that you can:

- Set the `defaultChecked` property, it will ensure that the checked state can be altered normally.
- Don't set the `isChecked` as it will make the input controlled.

```jsx static=true
<Switch
  type="checkbox"
  name="allow-cookies-uncontrolled"
  id="allow-cookies-uncontrolled"
  defaultChecked={true}
>
  Allow cookies
</Switch>
```

### Switch vs Checkbox

Switch is a two-step action: selection and execution, whereas checkbox is just selection of an option and its execution usually requires another control.

## Code examples

```jsx
<Switch id="testSwitch1" name="testSwitch1">
  label for switch
</Switch>
```

### Switch disabled

```jsx
<Switch id="testSwitch2" name="testSwitch2" isDisabled>
  label for switch
</Switch>
```

## Accessibility

- use clear and concise label for Switch component
- if needed provide additional information for the user if Switch will cause a change in the context

## Props

import { Props } from '@contentful/f36-docs-utils';

<Props of="Switch" />
