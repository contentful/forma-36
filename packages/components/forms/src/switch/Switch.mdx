---
title: 'Switch'
type: 'component'
status: 'stable'
slug: /components/switch/
github: 'https://github.com/contentful/forma-36/tree/master/packages/components/forms/src/switch'
storybook: 'https://f36-storybook.contentful.com/?path=/story/form-elements-switch--basic'
typescript: ./Switch.tsx
---

Switch is a control that is used to quickly switch between two possible states. Switch works like a physical switch that allows users to turn things on or off, like a light switch.

## Table of contents

- [How to use Switch](#how-to-use-switch)
- [Code examples](#code-examples)
- [Content recommendations](#content-recommendations)
- [Accessibility](#accessibility)

## How to use Switch

- use when a setting requires an on/off or show/hide function to display the results
- use when user needs to perform instantaneous actions that do not need a review or confirmation
- use when user is toggling independent features or behaviors

### Switch vs Checkbox

Switch is a two-step action: selection and execution, whereas checkbox is just selection of an option and its execution usually requires another control.

## Code examples

```jsx
<Switch id="testSwitch1" name="testSwitch1" label="label for switch" />
```

### Switch disabled

```jsx
<Switch
  id="testSwitch2"
  name="testSwitch2"
  label="label for switch"
  isDisabled
/>
```

## Accessibility

- use clear and concise label for Switch component
- if needed provide additional information for the user if Switch will cause a change in the context
