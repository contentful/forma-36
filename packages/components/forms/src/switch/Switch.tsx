import React from 'react';
import { BaseCheckbox, BaseCheckboxProps } from '../base-checkbox';
import { useFormControl } from '../form-control/FormControlContext';
import { ExpandProps } from '@contentful/f36-core';

export type SwitchProps = Omit<BaseCheckboxProps, 'type' | 'isIndeterminate'>;

const _Switch = (
  props: ExpandProps<SwitchProps>,
  ref: React.Ref<HTMLInputElement>,
) => {
  const {
    testId = 'cf-ui-switch',
    id,
    isDisabled,
    isInvalid,
    isRequired,
    children,
    ...otherProps
  } = props;

  // Removes the not needed properties that comes from FormControl context.
  const {
    inputValue,
    setInputValue,
    maxLength,
    setMaxLength,
    isReadOnly,
    ...formProps
  } = useFormControl({
    id,
    isDisabled,
    isInvalid,
    isRequired,
  });

  return (
    <BaseCheckbox
      {...formProps}
      {...otherProps}
      testId={testId}
      type="switch"
      ref={ref}
    >
      {children}
    </BaseCheckbox>
  );
};

export const Switch = React.forwardRef(_Switch);
