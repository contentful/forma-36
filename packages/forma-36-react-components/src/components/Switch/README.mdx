---
title: 'Switch'
type: 'component'
status: 'stable'
slug: /components/switch/
github: 'https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components/src/components/Switch'
storybook: 'https://f36-storybook.contentful.com/?path=/story/components-switch--default'
---

Detailed documentation will be provided soon. For now, please have a look into the storybook for implementation details.

```jsx
<Switch id="testSwitch" labelText="label" />
```
