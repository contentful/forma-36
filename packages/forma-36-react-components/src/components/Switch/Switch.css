@import 'settings/colors';
@import 'settings/dimensions';
@import 'settings/transitions';
@import 'settings/borders';
@import 'settings/shadows';

.Switch__wrapper {
  display: flex;
  position: relative;
  align-items: center;
}

.Switch {
  box-sizing: content-box;
  width: calc(1rem * (34 / var(--font-base-default)));
  height: calc(1rem * (16 / var(--font-base-default)));
  background-color: var(--color-text-lightest);
  border-radius: var(--border-radius-small);
  cursor: pointer;
  outline: none;
  margin: 0 var(--spacing-xs) 0 0;
  border: 2px solid var(--color-text-lightest);
  appearance: none;
  transition: background-color cubic-bezier(0.65, 0.05, 0.36, 1)
      var(--transition-duration-default),
    border-color cubic-bezier(0.65, 0.05, 0.36, 1)
      var(--transition-duration-default);

  &:focus {
    outline: none;
    border: 2px solid var(--color-text-lightest);
    box-shadow: var(--glow-primary);
  }
}

.Switch--small {
  width: calc(1rem * (28 / var(--font-base-default)));
  height: calc(1rem * (14 / var(--font-base-default)));

  & + .Switch__label {
    & .Switch__icon--done {
      left: 0.15rem;
    }

    & .Switch__icon--close {
      left: 1rem;
    }
  }
}

.Switch__label {
  font-family: var(--font-stack-primary);
  color: var(--color-text-dark);
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.Switch__icon {
  position: absolute;
  width: 0.85rem;
  height: 0.85rem;
  cursor: pointer;
  pointer-events: none;
}

.Switch:before {
  content: '';
  display: block;
  width: 1rem;
  height: 1rem;
  background-color: var(--color-white);
  transition: transform cubic-bezier(0.65, 0.05, 0.36, 1)
    var(--transition-duration-default);
  border-radius: 2px;
  transform: translate3d(0, 0, 0);
}

.Switch.Switch--small::before {
  width: 0.85rem;
  height: 0.85rem;
}

.Switch--checked {
  background-color: var(--color-primary);
  border-color: var(--color-primary);

  &:focus {
    outline: none;
    border: 2px solid var(--color-primary);
    box-shadow: var(--glow-primary);
  }
}

.Switch--disabled {
  background-color: var(--color-element-light);
  border-color: var(--color-element-light);
  cursor: not-allowed;

  &:before {
    background-color: var(--color-element-darkest);
  }
}

.Switch--checked.Switch--disabled {
  background-color: var(--color-element-mid);
  border-color: var(--color-element-mid);
}

.Switch--checked:before {
  transform: translate3d(113%, 0, 0);
}

.Switch--checked.Switch--small:before {
  transform: translate3d(105%, 0, 0);
}

.Switch__label--disabled {
  color: var(--color-text-light);

  & .Switch__icon {
    fill: var(--color-element-darkest);
  }
}

/* Temporary aligning icons, TODO in v4*/
.Switch__icon--done {
  left: 0.25rem;
}

.Switch__icon--close {
  left: 1.25rem;
}
