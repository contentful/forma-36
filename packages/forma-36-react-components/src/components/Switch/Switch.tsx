import React, { FunctionComponent, KeyboardEvent } from 'react';
import classNames from 'classnames';
import { Close, Done } from '@contentful/f36-icons';

import styles from './Switch.css';

export interface SwitchProps {
  onToggle?: (isChecked: boolean) => void;
  isChecked?: boolean;
  isDisabled?: boolean;
  labelText: string;
  size?: 'default' | 'small';
  className?: string;
  id: string;
}

export const Switch: FunctionComponent<SwitchProps> = (props: SwitchProps) => {
  return (
    <div className={classNames(styles['Switch__wrapper'], {}, props.className)}>
      <input
        type="checkbox"
        onClick={onToggle}
        onKeyUp={onKeyUp}
        defaultChecked={props.isChecked}
        disabled={props.isDisabled}
        className={classNames(styles['Switch'], {
          [styles['Switch--checked']]: props.isChecked,
          [styles['Switch--disabled']]: props.isDisabled,
          [styles['Switch--small']]: props.size === 'small',
        })}
        id={props.id}
      />
      <label
        className={classNames(styles['Switch__label'], {
          [styles['Switch__label--disabled']]: props.isDisabled,
        })}
        htmlFor={props.id}
      >
        <Done
          className={classNames(
            styles['Switch__icon'],
            styles['Switch__icon--done'],
          )}
          size="tiny"
          variant="white"
        />
        {props.labelText}
        <Close
          className={classNames(
            styles['Switch__icon'],
            styles['Switch__icon--close'],
          )}
          size="tiny"
          variant="white"
        />
      </label>
    </div>
  );

  function onToggle() {
    if (props.onToggle) {
      props.onToggle(!props.isChecked);
    }
  }

  function onKeyUp(e: KeyboardEvent<HTMLElement>) {
    if (props.onToggle && e.key === 'Enter') {
      props.onToggle(!props.isChecked);
    }
  }
};
