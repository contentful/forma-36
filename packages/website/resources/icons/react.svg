<svg height="46" viewBox="0 0 50 46" width="50" xmlns="http://www.w3.org/2000/svg"><g opacity=".119024"><path d="m49.9288618 23c0-3.4210526-4.1361789-6.6631579-10.4776423-8.6736842 1.4634146-6.69473685.8130081-12.02105264-2.0528455-13.7263158-.6605691-.4-1.4329268-.58947368-2.2764228-.58947368v2.34736842c.4674797 0 .843496.09473684 1.1585366.27368421 1.3821138.82105263 1.9817073 3.94736842 1.5142276 7.96842105-.1117886.9894737-.2947154 2.0315789-.5182926 3.0947368-1.99187-.5052631-4.1666667-.8947368-6.4532521-1.1473684-1.3719512-1.9473684-2.7947154-3.71578945-4.2276422-5.26315787 3.3130081-3.18947369 6.4227642-4.93684211 8.5365853-4.93684211v-2.34736842c-2.7947154 0-6.453252 2.06315789-10.152439 5.64210526-3.699187-3.55789473-7.3577236-5.6-10.152439-5.6v2.34736842c2.1036585 0 5.2235772 1.73684211 8.5365853 4.90526316-1.4227642 1.54736842-2.8455284 3.30526316-4.1971544 5.25263156-2.296748.2526316-4.4715447.6421053-6.4634147 1.1578948-.2337398-1.0526316-.406504-2.0736843-.5284553-3.0526316-.4776422-4.02105265.1117887-7.14736844 1.4837399-7.97894739.304878-.18947368.7012195-.27368421 1.1686992-.27368421v-2.34736842c-.8536586 0-1.6260163.18947368-2.296748.58947368-2.85569105 1.70526316-3.49593495 7.02105263-2.0223577 13.69473684-6.32113823 2.0210526-10.43699189 5.2526316-10.43699189 8.6631579 0 3.4210526 4.13617886 6.6631579 10.47764229 8.6736842-1.46341465 6.6947369-.81300814 12.0210526 2.0528455 13.7263158.6605691.4 1.4329268.5894737 2.2865854.5894737 2.7947154 0 6.453252-2.0631579 10.152439-5.6421053 3.699187 3.5578948 7.3577236 5.6 10.152439 5.6.8536586 0 1.6260163-.1894737 2.296748-.5894737 2.8556911-1.7052631 3.495935-7.0210526 2.0223577-13.6947368 6.300813-2.0105263 10.4166667-5.2526316 10.4166667-8.6631579zm-13.2317073-7.0210526c-.3760163 1.3578947-.843496 2.7578947-1.3719512 4.1578947-.4166667-.8421053-.8536586-1.6842105-1.3313009-2.5263158-.4674796-.8421052-.9654471-1.6631579-1.4634146-2.4631579 1.4430894.2210527 2.8353659.4947369 4.1666667.831579zm-4.6544716 11.2105263c-.7926829 1.4210526-1.605691 2.768421-2.449187 4.0210526-1.5142276.1368421-3.0487805.2105263-4.5934959.2105263-1.5345528 0-3.0691057-.0736842-4.5731707-.2-.843496-1.2526315-1.6666667-2.5894737-2.4593496-4-.7723577-1.3789473-1.4735773-2.7789473-2.1138212-4.1894737.6300813-1.4105263 1.3414635-2.8210526 2.1036586-4.2.7926829-1.4210526 1.605691-2.768421 2.449187-4.0210526 1.5142276-.1368421 3.0487805-.2105263 4.5934959-.2105263 1.5345528 0 3.0691057.0736842 4.5731707.2.843496 1.2526316 1.6666667 2.5894737 2.4593496 4 .7723577 1.3789474 1.4735773 2.7789474 2.1138212 4.1894737-.6402439 1.4105263-1.3414635 2.8210526-2.1036586 4.2zm3.2825204-1.3684211c.5487804 1.4105263 1.0162601 2.8210527 1.402439 4.1894737-1.3313008.3368421-2.7337399.6210526-4.1869919.8421053.4979675-.8105263.995935-1.6421053 1.4634146-2.4947369.4674797-.8421052.9044716-1.6947368 1.3211383-2.5368421zm-10.3048781 11.231579c-.9451219-1.0105263-1.8902439-2.1368421-2.8252032-3.3684211.9146341.0421053 1.8495934.0736842 2.7947154.0736842.9552846 0 1.9004065-.0210526 2.8252033-.0736842-.9146342 1.231579-1.8597561 2.3578948-2.7947155 3.3684211zm-7.5609756-6.2c-1.4430894-.2210527-2.8353659-.4947369-4.1666667-.831579.3760163-1.3578947.843496-2.7578947 1.3719512-4.1578947.4166667.8421053.8536586 1.6842105 1.3313009 2.5263158.4776422.8421052.9654471 1.6631579 1.4634146 2.4631579zm7.5101626-21.90526318c.9451219 1.01052632 1.8902439 2.13684208 2.8252032 3.36842108-.9146341-.0421053-1.8495934-.0736842-2.7947154-.0736842-.9552846 0-1.9004065.0210526-2.8252033.0736842.9146342-1.231579 1.8597561-2.35789476 2.7947155-3.36842108zm-7.5203252 6.19999998c-.4979675.8105263-.995935 1.6421053-1.4634146 2.4947369-.4674797.8421052-.9044716 1.6842105-1.3211383 2.5263158-.5487804-1.4105264-1.0162601-2.8210527-1.402439-4.1894737 1.3313008-.3263158 2.7337399-.6105263 4.1869919-.831579zm-9.19715448 13.1789474c-3.59756098-1.5894737-5.92479675-3.6736842-5.92479675-5.3263158s2.32723577-3.7473684 5.92479675-5.3263158c.87398374-.3894737 1.82926828-.7368421 2.81504068-1.0631579.5792683 2.0631579 1.3414634 4.2105263 2.2865853 6.4105263-.9349593 2.1894737-1.6869918 4.3263158-2.2560975 6.3789474-1.0060976-.3263158-1.96138214-.6842105-2.84552848-1.0736842zm5.46747968 15.0421053c-1.3821138-.8210527-1.9817073-3.9473685-1.5142276-7.9684211.1117886-.9894737.2947154-2.0315789.5182926-3.0947368 1.99187.5052631 4.1666667.8947368 6.4532521 1.1473684 1.3719512 1.9473684 2.7947154 3.7157895 4.2276422 5.2631579-3.3130081 3.1894737-6.4227642 4.9368421-8.5365853 4.9368421-.4573171-.0105263-.843496-.1052632-1.148374-.2842105zm24.1056911-8.0210527c.4776422 4.0210527-.1117887 7.1473684-1.4837399 7.9789474-.304878.1894737-.7012195.2736842-1.1686992.2736842-2.1036585 0-5.2235772-1.7368421-8.5365853-4.9052632 1.4227642-1.5473684 2.8455284-3.3052631 4.1971544-5.2526315 2.296748-.2526316 4.4715447-.6421053 6.4634147-1.1578948.2337398 1.0631579.4166666 2.0842106.5284553 3.0631579zm3.9126016-7.0210526c-.8739838.3894737-1.8292683.7368421-2.8150407 1.0631579-.5792683-2.0631579-1.3414634-4.2105263-2.2865853-6.4105263.9349593-2.1894737 1.6869918-4.3263158 2.2560975-6.3789474 1.0060976.3263158 1.9613821.6842105 2.8556911 1.0736842 3.597561 1.5894737 5.9247967 3.6736842 5.9247967 5.3263158-.0101626 1.6526316-2.3373983 3.7473684-5.9349593 5.3263158z"/><ellipse cx="24.989837" cy="23" rx="4.644309" ry="4.810526"/></g></svg>