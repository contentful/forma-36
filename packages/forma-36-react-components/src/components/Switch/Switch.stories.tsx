import React, { useState } from 'react';

import Switch, { SwitchProps } from './Switch';
import SectionHeading from '../Typography/SectionHeading';
import Flex from '../Flex/Flex';

export default {
  title: 'Components/Switch',
  component: Switch,
  parameters: {
    propTypes: [Switch['__docgenInfo']],
  },
  argTypes: {
    className: { control: { disable: true } },
    testId: { control: { disable: true } },
  },
};

export const Basic = (args: SwitchProps) => {
  const [isActive, setActive] = useState(false);

  return (
    <Switch
      {...args}
      id="testSwitch"
      isChecked={isActive}
      onToggle={setActive}
    />
  );
};

Basic.args = {
  labelText: 'My label text',
  size: 'default',
  isDisabled: false,
};

export const Overview = () => (
  <>
    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">Switch default</SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch id="testSwitch" labelText="My label text" />
    </Flex>
    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">Switch checked</SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch id="testSwitch" isChecked labelText="My label text" />
    </Flex>
    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">Switch disabled</SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch id="testSwitch" isDisabled labelText="My label text" />
    </Flex>

    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">Switch small</SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch size="small" id="testSwitch" labelText="My label text" />
    </Flex>
    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">Switch small checked</SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch
        size="small"
        id="testSwitch"
        isChecked
        labelText="My label text"
      />
    </Flex>
    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">
        Switch small checked disabled
      </SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch
        size="small"
        id="testSwitch"
        isDisabled
        isChecked
        labelText="My label text"
      />
    </Flex>
    <Flex marginBottom="spacingS">
      <SectionHeading element="h3">Switch small disabled</SectionHeading>
    </Flex>
    <Flex marginBottom="spacingS">
      <Switch
        size="small"
        id="testSwitch"
        isDisabled
        labelText="My label text"
      />
    </Flex>
  </>
);
