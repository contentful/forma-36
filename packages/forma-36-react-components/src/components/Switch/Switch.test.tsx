import React from 'react';
import { render } from '@testing-library/react';

import { Switch } from './Switch';

it('renders the component', () => {
  const { container } = render(<Switch id="testCheckbox" labelText="foobar" />);

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component as checked', () => {
  const { container } = render(
    <Switch id="testCheckbox" labelText="foobar" isChecked />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component as disabled', () => {
  const { container } = render(
    <Switch id="testCheckbox" labelText="foobar" isDisabled />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component as disabled and checked', () => {
  const { container } = render(
    <Switch id="testCheckbox" labelText="foobar" isDisabled isChecked />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const { container } = render(
    <Switch id="testCheckbox" labelText="foobar" className="my-extra-class" />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
