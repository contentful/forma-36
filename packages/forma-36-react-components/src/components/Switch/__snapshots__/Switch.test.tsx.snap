// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
.emotion-0 {
  display: inline-block;
  fill: #ffffff;
  height: 16px;
  width: 16px;
}

<div
  class="Switch__wrapper"
>
  <input
    class="Switch"
    id="testCheckbox"
    type="checkbox"
  />
  <label
    class="Switch__label"
    for="testCheckbox"
  >
    <svg
      class="Switch__icon Switch__icon--done emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
      />
    </svg>
    foobar
    <svg
      class="Switch__icon Switch__icon--close emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
      />
    </svg>
  </label>
</div>
`;

exports[`renders the component as checked 1`] = `
.emotion-0 {
  display: inline-block;
  fill: #ffffff;
  height: 16px;
  width: 16px;
}

<div
  class="Switch__wrapper"
>
  <input
    checked=""
    class="Switch Switch--checked"
    id="testCheckbox"
    type="checkbox"
  />
  <label
    class="Switch__label"
    for="testCheckbox"
  >
    <svg
      class="Switch__icon Switch__icon--done emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
      />
    </svg>
    foobar
    <svg
      class="Switch__icon Switch__icon--close emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
      />
    </svg>
  </label>
</div>
`;

exports[`renders the component as disabled 1`] = `
.emotion-0 {
  display: inline-block;
  fill: #ffffff;
  height: 16px;
  width: 16px;
}

<div
  class="Switch__wrapper"
>
  <input
    class="Switch Switch--disabled"
    disabled=""
    id="testCheckbox"
    type="checkbox"
  />
  <label
    class="Switch__label Switch__label--disabled"
    for="testCheckbox"
  >
    <svg
      class="Switch__icon Switch__icon--done emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
      />
    </svg>
    foobar
    <svg
      class="Switch__icon Switch__icon--close emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
      />
    </svg>
  </label>
</div>
`;

exports[`renders the component as disabled and checked 1`] = `
.emotion-0 {
  display: inline-block;
  fill: #ffffff;
  height: 16px;
  width: 16px;
}

<div
  class="Switch__wrapper"
>
  <input
    checked=""
    class="Switch Switch--checked Switch--disabled"
    disabled=""
    id="testCheckbox"
    type="checkbox"
  />
  <label
    class="Switch__label Switch__label--disabled"
    for="testCheckbox"
  >
    <svg
      class="Switch__icon Switch__icon--done emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
      />
    </svg>
    foobar
    <svg
      class="Switch__icon Switch__icon--close emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
      />
    </svg>
  </label>
</div>
`;

exports[`renders the component with an additional class name 1`] = `
.emotion-0 {
  display: inline-block;
  fill: #ffffff;
  height: 16px;
  width: 16px;
}

<div
  class="Switch__wrapper my-extra-class"
>
  <input
    class="Switch"
    id="testCheckbox"
    type="checkbox"
  />
  <label
    class="Switch__label"
    for="testCheckbox"
  >
    <svg
      class="Switch__icon Switch__icon--done emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
      />
    </svg>
    foobar
    <svg
      class="Switch__icon Switch__icon--close emotion-0"
      data-test-id="cf-ui-icon"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
      />
    </svg>
  </label>
</div>
`;
